src/ucp/rndv/rndv.c

```diff
@@ -994,12 +994,19 @@ ucp_rndv_send_frag_get_mem_type(ucp_request_t *sreq, size_t length,
 UCS_PROFILE_FUNC_VOID(ucp_rndv_recv_frag_get_completion, (self),
                       uct_completion_t *self)
 {
-    ucp_request_t *freq     = ucs_container_of(self, ucp_request_t,
-                                               send.state.uct_comp);
-    ucp_request_t *rndv_req = freq->super_req;
-    ucp_request_t *rreq     = rndv_req->super_req;
-    uint64_t offset         = freq->send.rndv.remote_address -
-                              rndv_req->send.rndv.remote_address;
+    ucp_request_t *freq = ucs_container_of(self, ucp_request_t,
+                                           send.state.uct_comp);
+    ucp_request_t *rndv_req, *rreq;
+    uint64_t offset;
+
+    if (freq->send.state.dt.offset != freq->send.length) {
```

Comment:
do we assume frag put will complete in order?

Reply:
we initiate them in order in `ucp_rndv_progress_rma_zcopy_common`
the check is similar to rndv get_zcopy flow in `ucp_rndv_get_completion`

Reply:
ok, it's because completions count starts from 0